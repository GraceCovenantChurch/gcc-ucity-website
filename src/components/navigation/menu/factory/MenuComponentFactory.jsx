import React from "react";
import ComponentIndex from "./MenuComponentDictionary";

const MenuComponentFactory = (props) => {
  const data = props.data;
  const ComponentToRender = ComponentIndex[data.type];

  return <ComponentToRender {...props} />;
};

export default MenuComponentFactory;
