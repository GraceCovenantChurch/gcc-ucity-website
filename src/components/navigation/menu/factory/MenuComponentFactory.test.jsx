import React from "react";
import "../../../../__mocks__/matchMedia";
import { Context as ResponsiveContext } from "react-responsive";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";

import MenuComponentFactory from "./MenuComponentFactory";

describe("MenuComponentFactory", () => {
  describe("Link", () => {
    const data = {
      title: "Family Groups",
      type: "link",
      children: "/familygroup",
    };

    test("matches the snapshot", () => {
      const { container: mobile } = render(
        <Router>
          <ResponsiveContext.Provider value={{ width: 300 }}>
            <MenuComponentFactory data={data} />
          </ResponsiveContext.Provider>
        </Router>
      );
      expect(mobile).toMatchSnapshot();

      const { container: desktop } = render(
        <Router>
          <ResponsiveContext.Provider value={{ width: 1000 }}>
            <MenuComponentFactory data={data} />
          </ResponsiveContext.Provider>
        </Router>
      );
      expect(desktop).toMatchSnapshot();
    });
  });

  describe("Dropdown", () => {
    const data = {
      title: "About",
      type: "dropdown",
      children: [{ title: "Welcome", type: "link", children: "/welcome" }],
    };

    test("matches the snapshot", () => {
      const { container: mobile } = render(
        <Router>
          <ResponsiveContext.Provider value={{ width: 300 }}>
            <MenuComponentFactory data={data} />
          </ResponsiveContext.Provider>
        </Router>
      );
      expect(mobile).toMatchSnapshot();

      const { container: desktop } = render(
        <Router>
          <ResponsiveContext.Provider value={{ width: 1000 }}>
            <MenuComponentFactory data={data} />
          </ResponsiveContext.Provider>
        </Router>
      );
      expect(desktop).toMatchSnapshot();
    });
  });
});
