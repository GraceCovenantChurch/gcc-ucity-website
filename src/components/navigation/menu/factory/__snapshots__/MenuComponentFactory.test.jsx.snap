// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuComponentFactory Dropdown matches the snapshot 1`] = `
<div>
  <button
    aria-haspopup="true"
    class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-dropdownButton-5 MuiButton-colorInherit"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        About
      </p>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`MenuComponentFactory Dropdown matches the snapshot 2`] = `
<div>
  <button
    aria-haspopup="true"
    class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-dropdownButton-5 MuiButton-colorInherit"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        About
      </p>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`MenuComponentFactory Link matches the snapshot 1`] = `
<div>
  <a
    aria-disabled="false"
    class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-menuButton-1"
    href="/familygroup"
    role="button"
    tabindex="0"
  >
    <span
      class="MuiButton-label"
    >
      <p
        class="MuiTypography-root makeStyles-menuText-2 MuiTypography-body1"
      >
        Family Groups
      </p>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </a>
</div>
`;

exports[`MenuComponentFactory Link matches the snapshot 2`] = `
<div>
  <a
    aria-disabled="false"
    class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-menuButton-1"
    href="/familygroup"
    role="button"
    tabindex="0"
  >
    <span
      class="MuiButton-label"
    >
      <p
        class="MuiTypography-root makeStyles-menuText-2 MuiTypography-body1"
      >
        Family Groups
      </p>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </a>
</div>
`;
